pylabnet/hardware/ni_daqs/nidaqmx_card.py

```diff
@@ -69,16 +93,99 @@ def _gen_ch_path(self, channel):
             channel=channel
         )
 
+
+class Service(ServiceBase):
+
+    def exposed_set_ao_voltage(self, ao_channel, voltage_pickle):
+        voltages = pickle.loads(voltage_pickle)
+        return self._module.set_ao_voltage(
+            ao_channel=ao_channel,
+            voltages=voltages
+        )
+
+
+class Client(ClientBase):
+
     def set_ao_voltage(self, ao_channel, voltages):
-        """Set analog output of NI DAQ mx card to a series of voltages
+        voltage_pickle = pickle.dumps(voltages)
+        return self._service.exposed_set_ao_voltage(
+            ao_channel=ao_channel,
+            voltage_pickle=voltage_pickle
+        )
 
-        :ao_channel: (str) Name of output channel (e.g. 'ao1', 'ao2')
-        :voltages: (list of int) list of voltages which will be output
+
+class GatedCounter(GatedCtrInterface):
+    """ Hardware class for NI gated counter """
+
+    def __init__(self):
+        self.ci_channel = None
+        self.task = None
+        self.duration = 0.1
+        self._status = 'Inactive'
+        self.current_count = 0
+
+    def activate_interface(self, counter_channel='Dev1/ctr0', physical_channel='/Dev1/20MHzTimebase'):
+        """ Activates counter interface (creates a task, does not start it)
+
+        :param counter_channel: (str) channel of counter to use, e.g. 'Dev1/ctr0'
+        :param physical_channel: (str) channel of physical counter input, e.g. 'Dev1/PFI0'
         """
 
-        # TODO: Understand the timing between ouput voltages (sample-wise?)
```

Comment:
Note: The Gated counter functionality is not yet complete. This should be addressed soon to enable NI card counting